clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -189,22 +189,33 @@ protected abstract void onJoinComplete(int generation,
      * Block until the coordinator for this group is known and is ready to receive requests.
      */
     public synchronized void ensureCoordinatorReady() {
+        ensureCoordinatorReady(0, Long.MAX_VALUE);
```

Comment:
Might be worth a comment mentioning that the use of 0 is intentional since the timeout is effectively infinite.